### ISSUE TITLE
Rule proposal: no-export-named-after-filename

### ISSUE THREAD

**Issue by edbrannin** [benmosher/eslint-plugin-import#1478]
Similar ideas have been floated in #1041 and #325, but this is a bit different:

If you have a Default export, don't also have a Named export that's named the same as the file.

This is to flag a possible root cause of [import/no-named-as-default](https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-named-as-default.md).

Bad:
```js
// Foo.jsx
import React from 'react'
import myHOC from './my-hoc'

const Foo = () => <p>Foo</p>

export default myHOC(Foo)
export { Foo } // For testing
```

Good:
```js
// Foo.jsx
import React from 'react'
import myHOC from './my-hoc'

const FooComponent = () => <p>Foo</p>

export default myHOC(Foo)
export { FooComponent } // For testing
```

Also good:
```js
// FooComponent.jsx
import React from 'react'

const FooComponent = () => <p>Foo</p>

export default FooComponent // OR export { FooComponent }
```
```js
// Foo.jsx
import React from 'react'
import FooComponent from './FooComponent'
import myHOC from './my-hoc'

export default myHOC(Foo)
```

Thoughts?

**Comment by ljharb** [comment]
Perhaps #1476 should also include an option that controls checking imports, exports, or both, and this issue’s suggestion could be the “exports” part? cc @golopot

**Comment by golopot** [comment]
Another similar rule proposal is `export-default-match-filename`, mentioned [here](https://github.com/benmosher/eslint-plugin-import/issues/325#issuecomment-280840161), 
 which reports cases like:
```js
// foo.js
export default function bar() {}
//                      ~~~  should match filename "foo.js"
```

I am not sure whether they should be separate rules, or all rolled into one.

**Comment by ljharb** [comment]
Having it rolled into one means the matching logic can be more centralized, both conceptually and technically.

**Comment by golopot** [comment]
I find it hard to name rule name and options when these are rolled into one. An attempt:
```js
{
  "default-name-match-filename": [2, {
    importDefaultName: true | {ignorePaths: string[]},
    exportDefaultName: true,
    noExportNameAfterFilename: true,
  }]
}
```
